src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -286,21 +309,18 @@ private function getProxyDumper()
      *
      * @return string
      */
-    private function addServiceLocalTempVariables($cId, Definition $definition, \SplObjectStorage $inlinedDefinitions, \SplObjectStorage $allInlinedDefinitions)
+    private function addServiceLocalTempVariables($cId, Definition $definition, \SplObjectStorage $inlinedDefinitions, array $serviceCalls, $preInstance = false)
     {
-        $allCalls = $calls = $behavior = array();
-
-        foreach ($allInlinedDefinitions as $def) {
-            $arguments = array($def->getArguments(), $def->getFactory(), $def->getProperties(), $def->getMethodCalls(), $def->getConfigurator());
-            $this->getServiceCallsFromArguments($arguments, $allCalls, false, $cId, $behavior, $allInlinedDefinitions[$def]);
-        }
+        $calls = array();
 
-        $isPreInstance = isset($inlinedDefinitions[$definition]) && isset($this->circularReferences[$cId]) && !$this->getProxyDumper()->isProxyCandidate($definition) && $definition->isShared();
         foreach ($inlinedDefinitions as $def) {
-            $this->getServiceCallsFromArguments(array($def->getArguments(), $def->getFactory()), $calls, $isPreInstance, $cId);
+            if ($preInstance && !$inlinedDefinitions[$def][1]) {
```

Comment:
Similar to the issue found in `ContainerBuidler`, `PhpDumper` also need to keep track of service reference that relate only to constructor arguments (of nested definitions). That's what `$inlinedDefinitions[$def][1]` does: if we're before the instantiation line of the service, and if the nested definition we're looking at is not involved in a constructor argument, it should be dumped after the instantiation line, thus `continue` here.